omnibus/files/private-chef-cookbooks/private-chef/templates/default/nginx/nginx_chef_api_lb.conf.erb

```diff
@@ -45,8 +45,30 @@
       alias /opt/opscode/version-manifest.txt;
     }
 
-    <% if node['private_chef']['data_collector']['proxy'] -%>
+    <% if node['private_chef']['data_collector']['root_url'] && node['private_chef']['data_collector']['token']-%>
     location "/data-collector/" {
+      access_by_lua '
+        ngx.req.read_body()
+        local token = "<%= node['private_chef']['data_collector']['token'] %>"
+
+        if ngx.var.request_method ~= "POST"
+        then
+          ngx.exit(ngx.HTTP_NOT_FOUND)
+        end
+
+        -- TODO: auth with /validate_request endpoint
+
+        local h = ngx.req.get_headers()
+
+        if h["x-data-collector-token"] == nil
```

Comment:
Could this lead to weird issues? For example, when the client sends a x-data-collector-token header with an invalid token, and [the check](https://github.com/chef/chef-server/pull/984/files#diff-e61ee1a4529c1a04df73e9febe4bbe70R59) passes, the request would still fail (because of that wrong token in the initial request). I'm not sure if there's a valid use case behind _not overriding the client's headers_ here, so I'd propose we just unconditionally set them (if the check passes, of course), so there's one possibility less to worry about when debugging data collector related issues...